src/GCDWheel.cs

```diff
@@ -269,51 +341,621 @@ public unsafe class GCDWheel {
+                if (!string.IsNullOrEmpty(remainingCastTimeString) && conf.castTimePosition == 1 && conf.CastTimeEnabled)
+                    ui.DrawCastBarText(remainingCastTimeString, combinedText, spellTimePos, conf.CastBarTextSize, true);
+            }
+        }
+
+        public class BarInfo {
+            public float CenterX { get; }
+            public float CenterY { get; }
+            public int Width { get; }
+            public int HalfWidth { get; }
+            public int RawHalfWidth { get; }
+            public int Height { get; }
+            public int HalfHeight { get; }
+            public int RawHalfHeight { get; }
+            public int BorderSize { get; }
+            public int HalfBorderSize { get; }
+            public int BorderSizeAdj { get; }
+            public float BorderWidthPercent { get; }
+            public float CurrentPos { get; }
+            public float GCDTime_SlidecastStart { get; } 
+            public float GCDTotal_SlidecastEnd { get; }
+            public float GCDTotal { get; }
+            public float CastTotal { get; }
+            public int TriangleOffset { get; }
+            public bool IsCastBar { get; } 
+            public bool IsShortCast { get; } 
+            public Vector2 StartVertex { get; }
+            public Vector2 EndVertex { get; }
+            public Vector2 ProgressVertex { get; }
+
+            public BarInfo(
+                float sizeX, 
+                float centX, 
+                float widthRatio, 
+                float sizeY, 
+                float centY, 
+                float heightRatio, 
+                int borderSize,
+                float castBarCurrentPos,
+                float gcdTime_slidecastStart, 
+                float gcdTotal_slidecastEnd,
+                int triangleOffset,
+                bool isCastBar, 
+                bool isShortCast) {
+
+                CenterX = centX;
+                CenterY = centY;
+                Width = (int)(sizeX * widthRatio);
+                HalfWidth = Width % 2 == 0 ? (Width / 2) : (Width / 2) + 1;
+                RawHalfWidth = Width / 2;
+                Height = (int)(sizeY * heightRatio);
+                HalfHeight = Height % 2 == 0 ? (Height / 2) : (Height / 2) + 1;
+                RawHalfHeight = Height / 2;
+                BorderSize = borderSize;
+                HalfBorderSize = BorderSize % 2 == 0 ? (BorderSize / 2) : (BorderSize / 2) + 1;
+                BorderSizeAdj = BorderSize >= 1 ? BorderSize : 1;
+                BorderWidthPercent = (float)BorderSizeAdj / (float)Width;
+                CurrentPos = castBarCurrentPos;
+                GCDTime_SlidecastStart = gcdTime_slidecastStart;
+                GCDTotal_SlidecastEnd = gcdTotal_slidecastEnd;
+                TriangleOffset = triangleOffset;
+                IsCastBar = isCastBar;
+                IsShortCast = isShortCast;
+                GCDTotal = DataStore.Action->TotalGCD;
```

Comment:
I think I fixed this in https://github.com/CaiClone/GCDTracker/commit/4bef91d988dbd35cbfb01003dad7609edaa8ec90

There are some other changes in there from the review as well.  I'll work on it some more tomorrow, but in the interim check and see if this fixes the issue for you.  The mount specifically might always be a bit jerky because the cast time is only 1s.  We are updating in visible chunks per iteration through the loop (as near as I can tell plugins aren't updated in parallel, so we have to wait for all other plugins to have their turn before we get to go again.

This bug would have effected more than just mounts: any interactions, teleports, etc.  Since it was quite a few things I think it best to handle this by creating an entirely new BarState.NonAbilityCast to handle it separately from the other cases.  It has some unique considerations (for example, the Queuelock should always be off).

We could catch this condition in the place you suggest, but I think it might be better to do it in as follows:

```
public void DrawCastBar (PluginUI ui) {  
   DrawBarElements(ui, true, gcdTotal > castTotal, **_gcdTotal < 0.001f**_, castbarProgress * castbarEnd, slidecastStart, slidecastEnd);  
}  
```

This has the downside of requiring an extra hard-coded false in DrawGCDBar:

```
public void DrawGCDBar(PluginUI ui) {  
    DrawBarElements(ui, false, shortCastFinished, false, gcdTime / gcdTotal, gcdTime, gcdTotal);  
}  
```

But I think it makes the most sense to catch it here.  We can then pass this bool onto BarInfo.IsNonAbility and it will be available there in case we need it in another context later.  What do you think?

Reply:
@CaiClone We need to decide if we should show the slidecast on non-ability casts.  DelvUI _does_ and I've emulated that behavior for now (since that's probably what most people are using who would be interested in using our castbar replacement)...

But it doesn't really make sense.  For some non-ability casts, you CAN move in the slidecast window and have the ability fire (like teleports).  In that context slidecast makes sense.  For things like mounts, either the slidecast window is very small, or is inconsistent; slidecasting doesn't work according to the mark we are making. 

In that case, slidecasting doesn't make sense at all.  _But_ DelvUI chooses to do the slidecasting anyway, even for mounts.

We could detect when the cast is a mount and disable it then, but there are probably other cases where slidecasting doesn't apply, and that would turn into a list of exceptions that might need curated with patches.  So I'd lean away from that, unless it is just mounts.

Reply:
I would be okay with doing it just for mounts. I think that's the big use case and I think it's worth it to disable it for that. I agree that a list of exceptions is the worst case, so unless we can neatly fit everything that doesn't allow slidecast into a recast group or we have a clean way to programmatically detect it just mounts sounds good.